.pre-commit-config.yaml

```diff
@@ -7,13 +7,12 @@ repos:
       - id: fix-byte-order-marker
 
   - repo: https://github.com/psf/black
-    rev: 23.3.0
+    rev: 23.9.1
```

Comment:
this version requires python 3.8
can you remove 3.7 from the the matrix in build.yaml?

Reply:
Hmm, is that the best idea, in that case there won't be any testing for 3.7. Unless we want to drop 3.7...

I think I can make some refactors to the ci to only run black for one pass.

Reply:
I'm not sure if it's worth adding complexity in order to support a python version that is not maintained anymore (tools like poetry and PDM also dropped 3.7).
Furthermore I already skipped CI testing of 3.7 on windows because of errors that I could not reproduce on my own windows laptop.

Reply:
Okay, cool. I'll try to address this today